spec/factories/purchases.rb

```diff
@@ -12,5 +12,13 @@
         user
       end
     end
+
+    trait :with_address do
```

Comment:
```suggestion
    trait :with_address do
      after(:build) do |purchase, evaluator|
        purchase.address_id = create(:address, user: purchase.user).id
      end
    end
```
でも本来は購入画面からの操作以外はaddress_idのvalidationをskipしたいところ